indico/modules/events/timetable/controllers/legacy.py

```diff
@@ -103,6 +107,77 @@ def _process(self):
+        elif self.timetable_entry.break_:
+            break_ = self.timetable_entry.break_
+            tt_entry_dt = self.timetable_entry.start_dt.astimezone(self.event_new.tzinfo)
+            form = BreakEntryForm(event=self.event_new, day=tt_entry_dt.date(),
+                                  obj=FormDefaults(break_, time=tt_entry_dt.time()))
+            if form.validate_on_submit():
+                # TODO: Check whether track_time_changes should be used.
+                update_break_entry(break_, form.data)
+                return jsonify_data(entries=[serialize_entry_update(break_.timetable_entry)], flash=False)
+        elif self.timetable_entry.session_block:
+            block = self.timetable_entry.session_block
+            tt_entry_dt = self.timetable_entry.start_dt.astimezone(self.event_new.tzinfo)
+            form = SessionBlockEntryForm(obj=FormDefaults(block, time=tt_entry_dt.time()),
+                                         event=self.event_new, session_block=block, to_schedule=False,
+                                         day=tt_entry_dt.date())
+            if form.validate_on_submit():
+                # TODO: Check whether track_time_changes should be used.
+                update_session_block(block, form.data)
+                return jsonify_data(entries=[serialize_entry_update(block.timetable_entry)], flash=False)
+        self.commit = False
+        return jsonify_form(form)
+
+
+class RHLegacyTimetableEditEntryTime(RHManageTimetableBase):
+    def _checkParams(self, params):
+        RHManageTimetableBase._checkParams(self, params)
+        self.timetable_entry = (self.event_new.timetable_entries
+                                .filter_by(id=request.view_args['timetable_entry_id'])
+                                .first_or_404())
+
+    def _process(self):
+        form = None
+        item = None
+        if self.timetable_entry.contribution:
+            item = self.timetable_entry.contribution
+        elif self.timetable_entry.break_:
+            item = self.timetable_entry.break_
+        tt_entry_dt = self.timetable_entry.start_dt.astimezone(self.event_new.tzinfo)
+        form = BaseEntryForm(obj=FormDefaults(item, time=tt_entry_dt.time()), day=tt_entry_dt.date(),
+                             event=self.event_new, item=item)
+        if form.validate_on_submit():
+            with track_time_changes():
+                if self.timetable_entry.contribution:
+                    update_timetable_entry(self.timetable_entry, {'start_dt': form.start_dt.data})
+                    update_contribution(item, {'duration': form.duration.data})
+                elif self.timetable_entry.break_:
+                    update_break_entry(item, form.data)
+            return jsonify_data(entries=[serialize_entry_update(item.timetable_entry)], flash=False)
+        self.commit = False
+        return jsonify_form(form, back_button=False, disabled_until_change=True)
```

Comment:
What's the problem? Shouldn't the form submit button be disabled until something changes in the form?